File: src/lib/ipc/events.ts
export const TRANSLATION_STARTED = 'translation:started' as const;
export const TRANSLATION_CHUNK = 'translation:chunk' as const;
export const TRANSLATION_COMPLETED = 'translation:completed' as const;
export const TRANSLATION_CANCELLED = 'translation:cancelled' as const;
export const TRANSLATION_ERROR = 'translation:error' as const;

export const MODEL_PULL_STARTED = 'model-pull:started' as const;
export const MODEL_PULL_PROGRESS = 'model-pull:progress' as const;
export const MODEL_PULL_COMPLETED = 'model-pull:completed' as const;
export const MODEL_PULL_ERROR = 'model-pull:error' as const;

export const POPUP_OPENED = 'popup:opened' as const;
export const POPUP_CLOSED = 'popup:closed' as const;
export const MENUBAR_OPENED = 'menubar:opened' as const;
export const MENUBAR_CLOSED = 'menubar:closed' as const;

export const NAV_REQUEST = 'nav:request' as const;

export type NavRoute = 'translate' | 'history' | 'settings';
export interface NavRequestPayload {
  route: NavRoute;
}

export type TranslationEvent =
  | typeof TRANSLATION_STARTED
  | typeof TRANSLATION_CHUNK
  | typeof TRANSLATION_COMPLETED
  | typeof TRANSLATION_CANCELLED
  | typeof TRANSLATION_ERROR;

export type ModelPullEvent =
  | typeof MODEL_PULL_STARTED
  | typeof MODEL_PULL_PROGRESS
  | typeof MODEL_PULL_COMPLETED
  | typeof MODEL_PULL_ERROR;

export type SurfaceEvent =
  | typeof POPUP_OPENED
  | typeof POPUP_CLOSED
  | typeof MENUBAR_OPENED
  | typeof MENUBAR_CLOSED
  | typeof NAV_REQUEST;

export type EventName = TranslationEvent | ModelPullEvent | SurfaceEvent;

File: src-tauri/src/events.rs
/// IPC 이벤트 이름 단일 정의.
/// FE 의 `src/lib/ipc/events.ts` 와 1:1 mirror. 양쪽 동시 변경.
pub const TRANSLATION_STARTED: &str = "translation:started";
pub const TRANSLATION_CHUNK: &str = "translation:chunk";
pub const TRANSLATION_COMPLETED: &str = "translation:completed";
pub const TRANSLATION_CANCELLED: &str = "translation:cancelled";
pub const TRANSLATION_ERROR: &str = "translation:error";

pub const MODEL_PULL_STARTED: &str = "model-pull:started";
pub const MODEL_PULL_PROGRESS: &str = "model-pull:progress";
pub const MODEL_PULL_COMPLETED: &str = "model-pull:completed";
pub const MODEL_PULL_ERROR: &str = "model-pull:error";

pub const POPUP_OPENED: &str = "popup:opened";
pub const POPUP_CLOSED: &str = "popup:closed";
pub const MENUBAR_OPENED: &str = "menubar:opened";
pub const MENUBAR_CLOSED: &str = "menubar:closed";

/// Tray menu 가 main 윈도우로 nav 전환을 요청. payload: `{ route: "translate" | "history" | "settings" }`.
pub const NAV_REQUEST: &str = "nav:request";
